### PR TITLE
Base django project

### PR DESCRIPTION
**Pull request by dimasciput**
Django project using : 
- Django version 2.0.2
- Python version 3.6

### CODE REVIEW

File: deployment/docker/REQUIREMENTS-dev.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+django-nose
+coverage
+pep8>=1.5.7,<1.6
+pylint
+flake8==3.3.0
+factory_boy
```
Comment (cchristelis) [inline deployment/docker/REQUIREMENTS-dev.txt]: I'm wondering why we would include some of the requirements here (like `factory_boy`) and not in production mode?

File: deployment/sites-enabled/default.conf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    # the port your site will be served on
+    listen      8080;
+    # the domain name it will serve for
+    server_name "";
+    charset     utf-8;
+
+    # Drop any non django related requests
+    # Its probably someone nefarious probing for vulnerabilities...
+    location ~ (\.php|.aspx|.asp|myadmin) {
```
Comment (cchristelis) [inline deployment/sites-enabled/default.conf]: thanks for adding this.

File: django_project/core/settings/base.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# coding=utf-8
+"""
+core.settings.base
+"""
+# Django settings for projecta project.
+
+from .utils import absolute_path
+
+ADMINS = (
+    ('Dimas Ciputra', 'dimas@kartoza.com'),
```
Comment (cchristelis) [inline django_project/core/settings/base.py]: Could you add @timlinux (?) and myself here as well, please. I'd like to be getting emails etc for the first while.

File: django_project/core/settings/prod_docker.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# noinspection PyUnresolvedReferences
+from .prod import *  # noqa
+import os
+
+DEBUG = False
+
+ALLOWED_HOSTS = ['*']
+
+ADMINS = (
+    ('Dimas Ciputra', 'dimas@kartoza.com'),)
```
Comment (cchristelis) [inline django_project/core/settings/prod_docker.py]: as above

File: deployment/docker/REQUIREMENTS-dev.txt (hunk 1 above)
Comment (dimasciput) [reply deployment/docker/REQUIREMENTS-dev.txt]: This is for testing purpose, that's why it's on the dev.

File: deployment/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# Pandoc needed to generate rst dumps, uic compressor needed for django-pipeline
+RUN apt-get update -y; apt-get -y --force-yes install yui-compressor
+
+RUN apt-get install -y python3-pip \
+    python-gdal \
+    python-geoip \
+    npm \
+    nodejs \
+    rpl \
```
[the remark this answers is not in the document]
Comment (dimasciput) [reply deployment/docker/Dockerfile]: I was following our old django base container https://hub.docker.com/r/kartoza/django-base/~/dockerfile/, you want me to remove these?

File: deployment/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+RUN dpkg-divert --local --rename --add /sbin/initctl
+
+# Pandoc needed to generate rst dumps, uic compressor needed for django-pipeline
+RUN apt-get update -y; apt-get -y --force-yes install yui-compressor
+
+RUN apt-get install -y python3-pip \
+    python-gdal \
+    python-geoip \
+    npm \
+    nodejs \
```
[the remark this answers is not in the document]
Comment (dimasciput) [reply deployment/docker/Dockerfile]: I was following our old django base container https://github.com/kartoza/docker-django-base/blob/master/Dockerfile , you want me to remove these?

File: deployment/docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# Note this base image is based on debian
+FROM python:3.6
+MAINTAINER Dimas Ciputra<dimas@kartoza.com>
+
+RUN export DEBIAN_FRONTEND=noninteractive
+ENV DEBIAN_FRONTEND noninteractive
+RUN dpkg-divert --local --rename --add /sbin/initctl
+
+# Pandoc needed to generate rst dumps, uic compressor needed for django-pipeline
+RUN apt-get update -y; apt-get -y --force-yes install yui-compressor
```
[the remark this answers is not in the document]
Comment (dimasciput) [reply deployment/docker/Dockerfile]: This is to compress the javascript, ok I will remove this for now.

File: deployment/docker/71-apt-cacher-ng.template [hunk 8]
```diff
@@ -0,0 +1,4 @@
+#Acquire::http { Proxy "http://192.168.0.106:3142"; };
```
[the remark this answers is not in the document]
Comment (dimasciput) [reply deployment/docker/71-apt-cacher-ng.template]: ok, will remove the cacher for now.

File: deployment/docker/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (cchristelis) [reply deployment/docker/Dockerfile]: I think whatever we don't need, we remove. Only adding it in again when we require it...

File: deployment/docker/REQUIREMENTS-dev.txt (hunk 1 above)
Comment (cchristelis) [reply deployment/docker/REQUIREMENTS-dev.txt]: Wouldn't we ever run tests in production mode on something like travis?

Comment (dimasciput) [reply deployment/docker/REQUIREMENTS-dev.txt]: In travis we are using dev mode.

File: django_project/core/settings/base.py (hunk 3 above)
Comment (dimasciput) [reply django_project/core/settings/base.py]: added

File: deployment/Makefile [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+	@echo "------------------------------------------------------------------"
+	@echo "Running in production mode"
+	@echo "------------------------------------------------------------------"
+	@docker-compose -p $(PROJECT_ID) up -d web
+	@# Dont confuse this with the dbbackup make command below
+	@# This one runs the postgis-backup cron container
+	@# We add --no-recreate so that it does not destroy & recreate the db container
+	@docker-compose -p $(PROJECT_ID) up --no-recreate --no-deps -d dbbackups
+
+permissions:
```
[the remark this answers is not in the document]
Comment (timlinux) [reply deployment/Makefile]: I think we dont really need permissions any more we can probably drop it...anyway we should store data in-volume

File: deployment/docker/REQUIREMENTS-dev.txt (hunk 1 above)
Comment (timlinux) [reply deployment/docker/REQUIREMENTS-dev.txt]: Yeah tests only run on the dev mode, and travis gets its own settings profile for testing.

**Pull request merged**